src/senselab/audio/tasks/text_to_speech/huggingface.py

```diff
@@ -0,0 +1,80 @@
+"""This module implements some utilities for the text-to-speech task."""
+
+from typing import Any, Dict, List, Optional
+
+from transformers import pipeline
+
+from senselab.audio.data_structures.audio import Audio
+from senselab.utils.data_structures.device import DeviceType, _select_device_and_dtype
+from senselab.utils.data_structures.model import HFModel
+
+
+class HuggingFaceTTS:
+    """A factory for managing Hugging Face TTS pipelines."""
+
+    _pipelines: Dict[str, pipeline] = {}
+
+    @classmethod
+    def _get_hf_tts_pipeline(
+        cls,
+        model: HFModel,
+        device: Optional[DeviceType] = None,
+    ) -> pipeline:
+        """Get or create a Hugging Face TTS pipeline.
+
+        Args:
+            model (HFModel): The Hugging Face model.
+            device (Optional[DeviceType]): The device to run the model on.
+
+        Returns:
+            pipeline: The TTS pipeline.
+        """
+        device, torch_dtype = _select_device_and_dtype(
+            user_preference=device, compatible_devices=[DeviceType.CUDA, DeviceType.CPU]
+        )
+        key = f"{model.path_or_uri}-{model.revision}-{device.value}"
+        if key not in cls._pipelines:
+            cls._pipelines[key] = pipeline(
+                "text-to-speech",
+                model=model.path_or_uri,
+                revision=model.revision,
+                device=device.value,
+                torch_dtype=torch_dtype
+            )
+        return cls._pipelines[key]
+
+    @classmethod
+    def synthesize_texts_with_transformers(
```

Comment:
Under the assumption you want people interacting with these through the API, then we should probably have this as _synthesize...?

Reply:
good point